### ISSUE TITLE
Public syslog server + signal tracking web page / API

### ISSUE THREAD

**Issue by madpsy** [richonguzman/LoRa_APRS_iGate#177]
Heads up that I've created a public syslog server which can ingest logs from an iGate. It lets the user view their own logs as well as plots data on maps - including signal strength etc. This is purely using the syslog output messages (does not have anything to do with aprs.fi !).

There's a rudimentary API and GeoJSON endpoint in case that's useful.

The website is: https://lora.link9.net/

Maybe you want to set 'lora.link9.net' port '1514' as the default for syslog settings (or the placeholder text?) - I don't mind if it was even enabled by default but that's up to you!

Any questions /concerns / comments please ask!

**Comment by IU5HKU** [comment]
Thank you so much, i'll give it a try

**Comment by IU5HKU** [comment]
There is something strange in the reports, i'm used to get long distance contacts, even more than the ones i see in your map, but this time these contacts are not mine, they aren't in the list of "Stations heard directly by" than you can read in aprs.fi website. Your site assign to me some packets digipeated by IQ5VK-11, that is the local club igate-digipeater. Ok, maybe i've igated some packets first heard by IQ5VK-11, but they can't be stated as i've heard directly it. If you want you can have a look at my reports for F5ZVD-4 , F1ZLL-4, 3A2ARM-10. Another strange thing is that in the "Stations who heard directly on radio" list of these callsign, you can't find IU5HKU-10 nor IQ5VK-11. I don't understand this behaviour. For the moment i disconnect my igate from your site, i don't wanna pollute the db with "strange" data, let me know. Anyway, i like the whole things, i find your service very useful.

**Comment by madpsy** [comment]
Thanks for testing it out. I knew there would be a few bugs and you found one!

Should be fixed now if you wouldn't mind enabling it again?

**Comment by IU5HKU** [comment]
Sure, the syslog is just enabled now, i'll keep it under control in the next hours/days, thank you for this great service :-)

**Comment by madpsy** [comment]
I see your traffic again - shall keep an eye on it.

Of course let me know if you are seeing too few or too many than you expect. I removed the problematic ones and hopefully my logic is right.

It will now ignore Digipeated packets as that isn't really the point of the service (maps etc) so I think this is sensible :-)

**Comment by richonguzman** [comment]
this is great for testing !!! nice!

**Comment by Mane76** [comment]
> There is something strange in the reports, i'm used to get long distance contacts, even more than the ones i see in your map, but this time these contacts are not mine, they aren't in the list of "Stations heard directly by" than you can read in aprs.fi website. Your site assign to me some packets digipeated by IQ5VK-11, that is the local club igate-digipeater. Ok, maybe i've igated some packets first heard by IQ5VK-11, but they can't be stated as i've heard directly it. If you want you can have a look at my reports for F5ZVD-4 , F1ZLL-4, 3A2ARM-10. Another strange thing is that in the "Stations who heard directly on radio" list of these callsign, you can't find IU5HKU-10 nor IQ5VK-11. I don't understand this behaviour. For the moment i disconnect my igate from your site, i don't wanna pollute the db with "strange" data, let me know. Anyway, i like the whole things, i find your service very useful.

Please consider the way APRS-IS is working - if one iGates receives a station it sends this info to APRS-IS and this is then displayed on the web - if multible iGates receive a station a filtering starts at the APRS-IS servers and only one (!) receiving iGate is displayed on the web. Usually the "fastest" one is displayed.
Now with syslog you see all (!) iGates which have received a single frame.

So should be pretty ok.

**Comment by IU5HKU** [comment]
Yes, for sure, but after the fix madpsy did, seems that infos are more suitable to the "real" situation, you can't consider a second-hop packet as directly heard by you...but now it works ok, stats are correct. Well done.

**Comment by S57PNX** [comment]
Can you add a button to collapse the left side of the screen (live logs) and make the map full size?

**Comment by madpsy** [comment]
@IU5HKU Glad the fix worked - you are still top for distance :-)

@richonguzman  It doesn't have a public code repo but there's an open API (https://lora.link9.net/api/readme.html) and a help page. 

I was thinking you could make  it the default host/port values for syslog settings (but off by default) and maybe a link to the site for convenience. Or... just link to the site in the readme/wiki and people can manually configure it if they want. Whatever you think is best for how you prefer to do things - I appreciate we haven't spoken before!

**Comment by madpsy** [comment]
@S57PNX  should be there now if you refresh

**Comment by S57PNX** [comment]
Thanks, looks very nice - but I can see the button in Chrome while I don't see it in Safari?!

I've added one of my gates (S58W-13) to your syslog to test, but I find some unexpected results - as if somehow digipeated trackers show as directly heard by my gate.

E.g.  S54Z-11 shows as being received by S58W-13, but I am sure it was digipeated by S55YLI-10. All these suspect cases show 
![Image 17  9  24 at 10 25](https://github.com/user-attachments/assets/2dd9c87b-5436-4c5c-8887-3671cac43894)

while the real direct contacts show this:

![Image 17  9  24 at 10 25](https://github.com/user-attachments/assets/e953bcbc-5933-43e1-9ad5-d0b8e959eb2c)

**Comment by S57PNX** [comment]
One more nitpick: in the pop-up bubble, the Heard: timestamp is off by one hour for me. A station that was just heard for the first time shows as Heard: 1h 0m. I am in GMT+1 observing DST, so effectively GMT+2.

**Comment by madpsy** [comment]
I knew that would happen - initially my intent for this was just for a UK club so didn't take other TZ's into account. Now I'll have to rethink a few things. Timezones are always an 'interesting' problem.

Have added 'ago' again (was there before but removed it - will see how it looks)

**Comment by madpsy** [comment]
Quick update. I've added an unauthenticated MQTT feed which might be useful to some folks.

Demo: https://lora.link9.net/mqtt-demo.html

**Comment by IU5HKU** [comment]
Hi madspy, there is something strange again in parsing the syslog, look:

HB4LO-9>APLRT1,IQ5VK-4*:!/6tx9P4UXO`GQHB4LO tracker Bat=4.05V (89%)

this is a packet digipeated by IQ5VK-4 and heard by my igate, IU5HKU-10. Your stats assign the contact with HB4LO-9 to my igate. Strange enough i can't find HB4LO-9 into IQ5VK-4 list of "directly heard stations" on aprs.fi, i can't say nothing about IQ5VK-4 syslog, i haven't access to it.

**Comment by madpsy** [comment]
Seems I committed a change and it was behind the main branch somehow so added my changes but reverted my previous commit to fix that!

Should be fixed now, for the second time

**Comment by richonguzman** [comment]
(I'm reading all of this, will return from Country Holidays on Monday to be online :P )

**Comment by S57PNX** [comment]
Another funny gltch - S58K-10 is plotted in a completely wrong spot, maybe 80km from its true location. 

Correct plot from aprs.fi:
![Image 20  9  24 at 18 20](https://github.com/user-attachments/assets/bbc134b9-f7f9-4a6f-9781-b14a42bad7f7)

Wrong plot on lora.link9.net:
![Image 20  9  24 at 18 19](https://github.com/user-attachments/assets/87fbca1c-a5f8-4d62-9ffe-68baa62c46cb)

**Comment by madpsy** [comment]
@S57PNX  If I look now it seems correct, right? Looking back in the logs its locations hasn't changed at all so how it managed to plot there is a mystery just now!

Let me know if you see it again though as that's a strange one.

**Comment by madpsy** [comment]
Actually still seeing it display wrong - only sometimes:

https://lora.link9.net/api/track?tx_callsign=S58K-10&points=1

That shows you the data it is plotting against. I wonder if this is a quirk of openstreetmap i'm unaware of. Shall do some digging.

**Comment by madpsy** [comment]
I think it was a bug with the `convertCoordinates()` function which was needed previously but not now.

If you can still make it break please use the 'Link' button at the top and paste the full link in here for me, thanks!

**Comment by S57PNX** [comment]
@madpsy thank you, it is looking correct now. I don't know how it happened to be, I just noticed it one day :)

Out of curiousity, what kind of resources do you have to run this database/web site? Will it handle an influx of new users?

**Comment by madpsy** [comment]
Just now it's running on a small VM (1 CPU / 4GB RAM) but is under no load:

`load average: 0.05, 0.01, 0.01`

If it becomes a problem can just throw more resource at it easily enough :slightly_smiling_face:  Just now the most resource intensive page is the 'Top 10' page. If that starts becoming slow I can add some caching  in various places so not overly concerned.

The MQTT page (now at http://lora-aprs.live) doesn't create any load as the processing is done on the client side. That is very much just a 'demo' for now as has a few known issues.

**Comment by madpsy** [comment]
![Screenshot_2024-09-23_17-43-51](https://github.com/user-attachments/assets/bcdf00ca-db46-40ef-8cef-702ffb527d79)

Have added SNR / distance / coverage heatmap for iGate owners, it's under the 'Coverage' link on the main page.

Example: https://lora.link9.net/coverage.html?rx_callsign=MB7UWI-10&time_period=48

**Comment by iw2ejh** [comment]
would it be possible to have the possibility to configure two syslog servers in the configuration, so we can have a shared public one, and a personal local one?

**Comment by richonguzman** [comment]
I guess I can add the second syslog posibility

but dont you think this would go on slowing more and more igate process?

can one syslog be linked into another?

**Comment by iw2ejh** [comment]
@richonguzman 
1-)but dont you think this would go on slowing more and more igate process?(
-- I honesty do not know what is the impact of sending log to syslog. You may valuate that based on the various task. If this took time and resources, obviously this function can be omitted. Better give priority to the original tasks.

2-)can one syslog be linked into another?(
-- I couldn't find an easy way to do that, some software are available but need installation and the data forwarding is not their  primary function. Standard Syslog servers usually do not have that capability.

**Comment by richonguzman** [comment]
> 2-)can one syslog be linked into another?( -- I couldn't find an easy way to do that, some software are available but need installation and the data forwarding is not their primary function. Standard Syslog servers usually do not have that capability.

usually syslog servers are more powerful than a ESP32 board this is why I asked ;)

**Comment by iw2ejh** [comment]
> usually syslog servers are more powerful than a ESP32 board this is why I asked ;)

If you have a dedicated sw running on a Server, yes, but at home we usually use a light app running on a NAS (in my case) that do not have any features other than recording data.

**Comment by madpsy** [comment]
Just FYI if you're sending syslogs to lora.link9.net there's a 'firehose' MQTT (JSON format) feed to read them back out again, example:

`mosquitto_sub -h hydros.link9.net -p 1883 -t lora_aprs/MB7UNH-10/logs | jq` 

That will show all logs for iGate MB7UNH-10. You can do all the usual MQTT filtering using + and # etc and the structure should be clear from the page below. I wrote a demo of a pure JS implementation using leaflet which only uses this feed to create a live map.

More info here:   https://lora-aprs.live/mqtt/

**Comment by S57PNX** [comment]
The problem with one syslog server forwarding to another server, or alternatively having a piece of software to split traffic to two syslog servers, is introducing a single point of failure. If this server or splitter fails, you lose both syslogs. 

This is why I would prefer to split at the source on ESP32, and have a possiblity to send to two syslogs. It's somewhere on my todo list, but won't have any time to work on my todo for multiple weeks.

**Comment by madpsy** [comment]
> One more nitpick: in the pop-up bubble, the Heard: timestamp is off by one hour for me. A station that was just heard for the first time shows as Heard: 1h 0m. I am in GMT+1 observing DST, so effectively GMT+2.

@S57PNX  Date/times in general should now be fixed throughout

**Comment by S57PNX** [comment]
Yes, time works ok now, tnx. And I like the new improvements (Moving etc.) However, one of my use cases for your web page is to track the perfarmance/behaviour/health of my gate(s), so I am curious which stations it is picking up. Can you add a filter to the map view to display only stations heard by a single gate? Like you can select the gate in the Coverage view. 

Currently, a tracker will "jump" between two gates that heard it, depending on which gate heard it last. I would rather see that "my" gate heard it 10 minutes ago, with an associated path and SNR etc. displayed, than some other gate heard it just now.

**Comment by madpsy** [comment]
Does this view not do that?

https://lora-aprs.live/igate.html?rx_callsign=S58W-13&time_period=20

or this view?

https://lora-aprs.live/history.html?rx_callsign=S58W-13&tx_callsign=S57PNX-14

or do you mean something else?

**Comment by S57PNX** [comment]
Oh yes, the first view ( https://lora-aprs.live/igate.html?rx_callsign=S58W-13&time_period=20 ) does exactly what I wanted, thanks. I just didn't realize that I needed to click on S58W-13 in the logs or on the maps to get this info. :)

**Comment by pdevlin13** [comment]
Hi madspy, thanks for doing this it is a really useful tool.
I have just noticed something that doesn’t seem right. In the screen shot attached the map shows MW0PDV-12 as receiving a beacon directly from MW0PDV-14. The beacon was actually sent via APRS and MW0PDV-12 received a digipeated packet from MW0PJE-12, see screen shot of my log.
![IMG_1100](https://github.com/user-attachments/assets/0ad65381-0c6b-4e1e-a015-11de268f84ff)
![IMG_1099](https://github.com/user-attachments/assets/cdebc402-eb8a-4f91-aeac-8d8552d216b7)

**Comment by pdevlin13** [comment]
Thanks for this madspy, it is a really useful tool. I thought I had a problem, but realised it was a beacon not showing on APRS but was in the log as Direct RX and so deleted that comment :-)
Thanks Paul
MW0PDV

**Comment by madpsy** [comment]
Glad to hear - if you do notice anything wrong obviously let me know though - it's quite a complicated beast now with many opportunities for bugs!

**Comment by madpsy** [comment]
Have made a 'beta' frontend page for tracking elevation vs height vs distance from iGate. Uses the public 'track' API endpoint so all of this is purely frontend code you can see via 'view source' and feel free to improve! Unsure if it's interesting but I thought it was kind of cool to see. Change tx_callsign to whatever you want. Example:

https://lora-aprs.live/elevation.html?tx_callsign=GM0UDL-7

'points' can be changed via the arrows

**Comment by S57PNX** [comment]
And when it comes to iGate Unique trackers, S58W-13 is the only gate listed for Croatia, even though it is not located in Croatia.

**Comment by madpsy** [comment]
@S57PNX Yeah it's based on the origin of the packet i.e. the Tracker. Have added a note at the top. I know it's a bit confusing so open to ideas!

**Comment by madpsy** [comment]
Ah ok - there's maybe a bordering issue with the naturalearthdata.com dataset or a bug in my implementation. It's really difficult to get right as I've learned whilst doing this.

This uses the same location service which I wrote using that data -  it's a lot more complicated to do than I originally thought!

https://lora-aprs.live/plotter.html

**Comment by S57PNX** [comment]
Yeah, the plotter has the same offset. S59HIJ-7 is in reality around 4km away from the Croatian border, yet your service shows it in Croatia. I did some spot checks around the country and there seems to be a random 0-4km offset from the real border. The nearest city feature works reasonably OK, so I simply believe that the border data is bad.

**Comment by madpsy** [comment]
That should be a bit better now, though won't be perfect due to the resolution. Hopefully improved though.

**Comment by madpsy** [comment]
@S57PNX The top lists are now by country of the iGate and hopefully the country accuracy is improved (also a new list for 'countries' on the top 10)

**Comment by S57PNX** [comment]
Thanks, the country accuracy is indeed improved. But I can still see up to 1km around the border attributed incorrectly to one or the other country. Doesn't hurt anything right now, but technically it is still not correct. 

Out of curiosity, how do you do it? Do you have country borders as geographic coordinates and then you plot polygons with them?

**Comment by madpsy** [comment]
Rather than talk endlessly about how it works (and bore everyone in the process) I'll put the country lookup service on my Github along with a test.html. Ff you want to look at / improve it then feel free. I am by no means an expert on the subject but was a fun learning experience.

**Comment by madpsy** [comment]
@S57PNX  https://github.com/madpsy/country-lookup

**Comment by richonguzman** [comment]
@madpsy its great page , I'm uploading data for a day and a half buy I see not digirepeated stations or others that are not with my callsign

Example:

CA2RXU-10 (igate) hears XQ3PCN-3 (digi) packet which is repeated by another digi (CA2RXU-11)

on packets listened and uploaded to syslog it shows the XQ3PCN-3 packet, but its not showing on map. Should it appear ?

**Comment by madpsy** [comment]
@richonguzman A digipeated packet doesn't contain anything useful in terms of RF performance I don't think. It may contain its location but it wouldn't have any SNR/RSSI data so I discard it. 

In your example I would have RF data for CA2RXU-11 but not XQ3PCN-3 but please do correct me if I'm wrong!

**Comment by richonguzman** [comment]
> @richonguzman A digipeated packet doesn't contain anything useful in terms of RF performance I don't think. It may contain its location but it wouldn't have any SNR/RSSI data so I discard it.
> 
> In your example I would have RF data for CA2RXU-11 but not XQ3PCN-3 but please do correct me if I'm wrong!

you are right, but why is not (only)  position useful ?

**Comment by madpsy** [comment]
My idea wasn't to re-create aprs.fi etc but to focus on the RF side of things, which none of those sites do. I could add purely positional data from digipeated packets if that's useful and make SNR/RSSI unknown or something.

Also, if your igate (CA2RXU-10) heard a direct digipeated packet it would have SNR/RSSI data but probably no positioning info in the packet. I wanted to be careful not to reinvent what existing ARPS sites already do.

**Comment by richonguzman** [comment]
well the digirepeater , although not hear directly, was heard being repeated over RF. (it may not have SNR / RSSI data but its still useful as its nearby stations heard by RF)

what do you think?

**Comment by madpsy** [comment]
`<165>1 - CA2RXU-10 CA2RXU_LoRa_iGate_1.3 - - - RX / GPS / XQ3PCN-3 / APLRG1,CA2RXU-11* / - / -59dBm / 7.50dB / -3591Hz / -32.97617N / -71.01334E / 52.7km / DIGI LORA CERRO EL ROBLE`

XQ3PCN-3 -> CA2RXU-11 -> CA2RXU-10 is the RF path

-32.97617N / -71.01334E is XQ3PCN-3's location and 52.7km is the distance between XQ3PCN-3 & CA2RXU-11 
-59dBm / 7.50dB is the signal from CA2RXU-11 -> CA2RXU-10

is that right?

**Comment by richonguzman** [comment]
> `<165>1 - CA2RXU-10 CA2RXU_LoRa_iGate_1.3 - - - RX / GPS / XQ3PCN-3 / APLRG1,CA2RXU-11* / - / -59dBm / 7.50dB / -3591Hz / -32.97617N / -71.01334E / 52.7km / DIGI LORA CERRO EL ROBLE`
> 
> XQ3PCN-3 -> CA2RXU-11 -> CA2RXU-10 is the RF path
> 
> -32.97617N / -71.01334E is XQ3PCN-3's location and 52.7km is the distance between XQ3PCN-3 & CA2RXU-11 -59dBm / 7.50dB is the signal from CA2RXU-11 -> CA2RXU-10
> 
> is that right?

the distance es from XQ3PCN-3(digi) and CA2RXU-10 (igate): the distance is calculated from the GPS position of the station, not from the repeater (CA2RXU-11)

**Comment by madpsy** [comment]
Got it, ok, given I have the last known location of CA2RXU-11 I can work out the distance between it and XQ3PCN-3 anyway.

**Comment by madpsy** [comment]
It's a fair amount of changes to make that happen but will take a look this weekend. Have 7 (hopefully easy) bugs to fix first in the list. 

Off topic really but how do I go about decoding the new encoded battery data? I couldn't make any sense of it! I have the old 'comment' based battery data working but the ARPS encoded one I gave up trying to decipher:

https://lora-aprs.live/battery.html

**Comment by richonguzman** [comment]
> It's a fair amount of changes to make that happen but will take a look this weekend. Have 7 (hopefully easy) bugs to fix first in the list.
> 
> Off topic really but how do I go about decoding the new encoded battery data? I couldn't make any sense of it! I have the old 'comment' based battery data working but the ARPS encoded one I gave up trying to decipher:
> 
> https://lora-aprs.live/battery.html

Base91 enconding : [Link of pdf APRS12B](https://github.com/wb2osz/aprsspec/blob/main/APRS12b.pdf)

page 68 at the bottom

**Comment by madpsy** [comment]
From what I remember in the syslog ouput they are wrapped in pipes '|<data>|' I presume the data is actually between them and the pipes are something you added

**Comment by richonguzman** [comment]
> From what I remember in the syslog ouput they are wrapped in pipes '||' I presume the data is actually between them and the pipes are something you added

"|" + 2byteA + 2bytesB + "|"
"|" + 2byteA + "|"
"|" + 2bytesB + "|"

2bytesA could be internal battery voltage
2bytesB could be external voltage

watch out as this values depends on the 3 pre-Telemetry packets definitions, UNIT, PARM, EQNS

**Comment by madpsy** [comment]
Thanks very much for the info

**Comment by S57PNX** [comment]
@madpsy can you do something about digis? Currently, I see them considered as trackers, whereas they belong more to the gates - or ideally even as a third class of the object. Not sure if syslog contains enough info to recognize digis as such though.

S55YAR-12 is a classic example of a pure digi. Yet under stats and on map it gets considered as a tracker.

**Comment by S57PNX** [comment]
and secondly, is there any heuristics to detect and remove impossibly long links? I can spot at least one 1000km+ link that in reality should be more like 10km, based on the geography.

**Comment by madpsy** [comment]
Don't think I know the difference between a true 'tracker' and a digipeter. Regarding the distance issue, it's a strange one. Clearly the tracker has sent those co-ordinates for whatever reason. Just now it's limited to 2000 km. If you check this link you'll see any which that has caught already:

https://lora-aprs.live/logs.html?code=106&page=1

The question is what limit is sensible without excluding long distances from things like balloons etc.

**Comment by madpsy** [comment]
This is far more complicated than I thought. Preview with so many bugs:

https://lora-aprs.live/map2.html

It isn't correct but it is progress in terms of something digipeating related!

**Comment by richonguzman** [comment]
> This is far more complicated than I thought. Preview with so many bugs:
> 
> https://lora-aprs.live/map2.html
> 
> It isn't correct but it is progress in terms of something digipeating related!

nice!

iGates appear as the Digi it repeated, not where it entered APRS-is /Syslog 

but all looks good in "my place"

**Comment by madpsy** [comment]
Yup, XQ3PCN-3 should be via CA2RXU-11, not  -6 

This side of things is all HTML/JS so can see the source for the page. I am by  no means a frontend dev so is kind of painful if I'm honest!

**Comment by richonguzman** [comment]
> Yup, XQ3PCN-3 should be via CA2RXU-11, not -6
> 
> This side of things is all HTML/JS so can see the source for the page. I am by no means a frontend dev so is kind of painful if I'm honest!

got cha! I have to learn JS and HTML also for this firmware.... so keep up the good work 🥇

**Comment by madpsy** [comment]
at least XQ3PCN-3 is showing now for you, although not via the correct path. Work in progress :sl

**Comment by richonguzman** [comment]
> at least XQ3PCN-3 is showing now for you, although not via the correct path. Work in progress :sl

this is turning into a super good tool ;)

this will help to bring better networks of pure RF paths to cover cities

**Comment by madpsy** [comment]
The goal was always to be interesting from an RF point of view rather than just location, hence why your SNR/RSSI syslogs are very useful. Digipeating is different though - I have no plans to go beyond one-hop for this

**Comment by richonguzman** [comment]
> The goal was always to be interesting from an RF point of view rather than just location, hence why your SNR/RSSI syslogs are very useful. Digipeating is different though - I have no plans to go beyond one-hop for this

1 hop is like 99% of the repeated path for LoRa. So it will be fine ;)

**Comment by madpsy** [comment]
In that case I will get the path right for 1-hop and call it done in that regard. Busy with work but hopefully by the weekend.

**Comment by richonguzman** [comment]
> In that case I will get the path right for 1-hop and call it done in that regard. Busy with work but hopefully by the weekend. Interest has been gathering pace:


iGates has grown 700 in 100 last days, Trackers 350 in same period.

**Comment by madpsy** [comment]
Once this supports 1-hop digipeaters I was wondering if you wanted to make it the 'placeholder' for syslog text input field (not enabled by default) - depends how reliability etc for sure

**Comment by richonguzman** [comment]
> Once this supports 1-hop digipeaters I was wondering if you wanted to make it the 'placeholder' for syslog text input field (not enabled by default) - depends on reliability etc for sure

sure thing!

**Comment by madpsy** [comment]
My brain now hurts! If anyone fancies taking a quick look at:

https://lora-aprs.live/map2.html

Just to ensure it makes sense and vaguely correct that would be appreciated.

I underestimated how complex it would be supporting digipeaters but I think i'm along the right lines now.

Maybe @richonguzman and @S57PNX  just if/when you have a few minutes spare?

**Comment by S57PNX** [comment]
Sorry I've been AFK for the last couple of days, I am just checking it now. At first glance it is a definite improvement, presuming color blue is the digi. But I see too much blue now - I suspect it is igates that operate as igate, but send a beacon over RF as well. That beacon gets picked up by another gate, which makes the originating gate blue.

I haven't really thought this over though and I don't see a good solution. @richonguzman  any ideas?

**Comment by madpsy** [comment]
@S57PNX  Thanks for the feedback - do you have an example callsign which is showing as a digipeater (blue) but is actually an iGate?

It could just be the iGate isn't sending syslog to the site - in which case if it did then it should turn green and show as an iGate (just guessing until you give an example)

**Comment by S57PNX** [comment]
You are right, none of the blue gates are currently sending syslog to you - which explains it. I'll reconfigure a few over weekend to send syslogs and see if they become green.

I am curious about S59DEM-11 though. It's green but I am fairly certain it is not sending a syslog to your site (unless somebody went there and changed the config without me knowing).

Can you add a tickbox on top to filter on blue objects?

**Comment by madpsy** [comment]
Found a bug which explains why S59DEM-11 was showing green. It looks good now (note the `object_type`). Yeah I'll be adding new filters for digipeaters

`tx_callsign	"S59DEM-11"
received_at	"2024-11-02T10:54:42+00:00"
rx_callsign	"S58W-13"
destination	"APLRG1"
path	"WIDE1-1"
signal_strength	-103
signal_quality	1.75
frequency_offset	-2633
latitude	"45.7582"
longitude	"14.29483"
distance	35.3
battery	4.25
comment	"LoRa APRS Batt=4.25V"
elevation	1257
country_code	"SI"
rx_igate	"S58W-13"
digipeater_distance	null
object_type	"digipeater"`

**Comment by S57PNX** [comment]
Looking great!

One more nitpick: ticking on Moving grays out igate and tracker tickboxs, but leaves them enabled (which was probably not the intention). I'd assume that Moving means only moving objects and igate and tracker need to be switched off? Or at least igate (dependoing on the background logic).

**Comment by madpsy** [comment]
It was fetching the wrong endpoint for moving objects so didn't refresh the map - fixed now

**Comment by S57PNX** [comment]
@madpsy another feature request :) Can you add a toggle for displaying the station labels? Many stations in a small country make for a very congested map. It would be great if we could toggle off the labels, leave just the position dots and keep the mouseover menu.

<img width="1326" alt="image" src="https://github.com/user-attachments/assets/b26af052-8ea4-4b57-a41f-6aeed4178bf5">

**Comment by madpsy** [comment]
@S57PNX  Your network is clearly too big! Kidding of course. The main reason for the chaos there is because, unlike APRS-IS, I don't deduplicate the digipeated packets so you see every node involved. That's on purpose though.

Getting back to your actual request - have added a Minimal view. Looks a bit crap but hopefully should do for now.

**Comment by S57PNX** [comment]
I perfectly understand why we get to see every node involved, and we thank you for that - that's exactly the purpose of parsing the syslogs in comparison to APRS-IS based web pages. However, you ain't seen nothing yet, there's another 50 or so igates in our network yet to start sending data to you!

Thanks for the minimal view, looks much better.

**Comment by madpsy** [comment]
Lightweight Python terminal application to view iGate logs in real time. Nothing fancy for a proof of concept, feel free to fork and do whatever you want with it :-)

https://github.com/madpsy/lora-aprs-python-client

![main](https://github.com/user-attachments/assets/5c00f53c-4c22-4234-8b03-e21de334960d)
![select](https://github.com/user-attachments/assets/1ae9b315-1bad-4d68-8721-5cd37b56dec3)

**Comment by madpsy** [comment]
Also a web equivalent:

https://lora-aprs.live/mqtt/stream.html?igate=S58W-13

Making them look/feel the same was more fun rather than a promise :-)

![web](https://github.com/user-attachments/assets/cd6a8b2c-874e-46e7-a145-b7e7ca4f6f18)

**Comment by madpsy** [comment]
Linux and Windows binaries available:

https://github.com/madpsy/lora-aprs-python-client/releases

**Comment by S57PNX** [comment]
Hi @madpsy , I got a funny one for you... My friends were all excited this morning about an amazing performance of the gate S55DLM-10, doing 200+ km through a mountain range. Too good to be true. So I investigated and this is what I believe happend.

S57C-9 (the tracker on the other side of the mountain) has most llikely two trackers with the same SSID. One is at his home beaconing via RF and is picked by S55DLM-10 some 18km away. The other tracker is AprsDroid via TCPIP, 200+ km away. Clearly this creates confusion and should not be done. However, your map also gets confused and seems to take the RF report of the near tracker with the location of the far tracker, showing an impossibly long RF path. 

<img width="357" alt="image" src="https://github.com/user-attachments/assets/f0fa9abf-827c-4696-93b3-4cf31cda9713">

<img width="1422" alt="image" src="https://github.com/user-attachments/assets/6a8f17a0-0b68-4fad-b0d4-48b85251e294">


Ah, I think I might have found the smoking gun:

This is the RF tracker at home that the gate is receiveng via RF:

<img width="668" alt="image" src="https://github.com/user-attachments/assets/8c4a8101-66b3-4f8e-b05f-6c499b4d2d80">

And this is the AprsDroid tracker 200km away - but why is it logged by the gate and considered a Direct RX by your server?

<img width="673" alt="image" src="https://github.com/user-attachments/assets/bb8b7abc-65b3-4043-851a-8e4fc9bf702c">

I think the gate gets the aprsdroid report from aprs-is but because it was also seen locally via RF within the last 30 minutes, it will be syslogged and reported back to APRS-IS as "heard". @richonguzman , does this make sense?

A quick fix would be to  filter out any packets with TCPIP* in the path from Direct RX category.

**Comment by madpsy** [comment]
@S57PNX  Nice find - indeed those should not be direct RX - have fixed that now!

**Comment by S57PNX** [comment]
Thank you, it looks better now and some other less obvious duplicates are gone too.

**Comment by richonguzman** [comment]
when syslog sends an aprsdroid packet (with APRSdroid TOCALL) it is received by LoRa through RF. TOCALLs are not changed when sended by tracker as tracker or TNC station.

**Comment by S57PNX** [comment]
User has same ssid on RF tracker at home and on aprsdroid 200km away. In this case the gate received an aprsdroid packet through APRS-IS and it was logged in syslog as received.

Why is the gate logging a packet with path TCPIP* in his syslog since it came from APRS-IS?

I think this happened because the same SSID was in RF range less than 30 minutes ago, so the gate had this call (with same SSID) in history and accepted it.

**Comment by richonguzman** [comment]
so it the "same callsign+ssid" error?

**Comment by S57PNX** [comment]
Yes, but as a result of two trackers with same callsign+ssid we get this in the gate log:

![Image](https://github.com/user-attachments/assets/be0961f0-0a27-41fd-a146-84554fc2aafb)

Should gate really log a packet that came from APRS-IS ? And why is there a RSSI and S/N on a packet that originated from aprsdroid and was sent directly to APRS-IS? I think the gate was confused because it saw an earlier packet from the duplicate tracker via RF and mixed both together.

**Comment by richonguzman** [comment]
iGate does not log (to syslog) packets listened from APRS-IS unless is a query-answer.

this may be a third party packet over RF.

**Comment by S57PNX** [comment]
It is almost impossible that this aprsdroid packet came over RF. Aprsdroid is behind a big mountain 200km away. But a real tracker with same call+ssid is 18km away and received normally in the syslog.

**Comment by ohanssen** [comment]
Hi, Great work. I have an issue thogh: Packets with timestamp are not accepted. Marked as Invalid GPS Type. The APRS Packet Decoder gets it correctly though.

**Comment by richonguzman** [comment]
Hi

This timestamp is faulty on my igate firmware step, right?

I totally missed to decode those packets with timestamp

**Comment by ohanssen** [comment]
Ah. Thanks. That may explain it :)
